app/components/3dmol/3dmol.directive.js

```diff
@@ -1,106 +1,246 @@
+                    $scope.animationPromise = null;
+                }
+
+                $scope.mol = Molecule.getByInchiKey({moleculeId: inchikey}, function(mol) {
+                    $scope.viewer.clear();
+                    $scope.displayMolecule(mol, $scope.style);
+                });
+            };
+
+            $scope.showMolecule = function() {
+                $state.go('molecule', {moleculeId: $scope.selectedMolecule.inchikey });
+            };
+
+            $scope.hasAnimation = function() {
+                return $scope.selectedMolecule && $scope.selectedMolecule.inchikey === 'RYYVLZVUVIJVGH-UHFFFAOYSA-N';
+            };
+
+            $scope.isAnimating = false;
+
+            $scope.animateMolecule = function() {
+
+                if ($scope.isAnimating) {
+
+                    if ($scope.animationPromise) {
+                        $interval.cancel($scope.animationPromise);
+                        $scope.animationPromise = null;
+                    }
+                    $rootScope.$broadcast('mongochem-frequency-histogram-selectbar', -1);
+                    $scope.isAnimating = false;
+                }
+                else {
+                    $scope.isAnimating = true;
+
+                    let animate = function() {
+                        $scope.animationPromise = $interval(function() {
+                            $scope.models[$scope.currmol].setStyle({},{hidden:true});
+                            $scope.currmol = ($scope.currmol+1) % $scope.models.length;
+                            $scope.models[$scope.currmol].setStyle({}, $scope.style);
+                            $scope.viewer.render();
+                            $rootScope.$broadcast('mongochem-frequency-histogram-selectbar', $scope.currmol);
+                        }, 100);
+                    };
+
+                    $scope.hasAnimationData = true;
+                    if (!$scope.models) {
+                        $scope.currmol = 0;
+                        $.get('/data/caffeine.json').
+                            success(function(data) {
+                                $scope.models = [];
+
+                                $scope.viewer.removeAllModels();
+                                angular.forEach(data.deltas, function(delta) {
+                                    var model = $scope.viewer.createModelFrom({serial: -1}, false);
+                                    $scope.models.push(model);
+
+                                    var atoms = angular.copy(data.atoms);
+
+                                    // Now add the delta to atom positions
+                                    for(var i = 0; i < atoms.length; i++) {
+                                        atoms[i].model = model.getID();
+                                        atoms[i].color = $3Dmol.elementColors.rasmol[atoms[i].elem];
+                                        atoms[i].x += delta[i][0];
+                                        atoms[i].y += delta[i][1];
```

Comment:
Good catch, my test data only had changes in x and y so guess I didn't bother with z, I will push a fix.